superset/viz.py

```diff
@@ -269,7 +269,7 @@ def query_obj(self):
         """Building a query object"""
         form_data = self.form_data
         groupby = form_data.get("groupby") or []
-        metrics = form_data.get("metrics") or []
+        metrics = form_data.get("metrics") or ['count']
```

Comment:
This is a mostly-safe assumption as we create that metric on behalf of the user. Though a user could remove that metric, of a new datasource type may not know that it has to create a count by default.

I had to dealt with this while deprecating v1 as defaults are now set on the client side and decided to take the first metric from the list of possible metric for the datasource.

I'm fine either way, just wanted to chime in.